libdnf/module/module_query.cpp

```diff
@@ -100,4 +100,44 @@ void ModuleQuery::filter_arch(const std::vector<std::string> & patterns, libdnf:
 }
 
 
+bool ModuleQuery::latest_cmp(const ModuleItem * module_item_1, const ModuleItem * module_item_2) {
+    Pool * pool = module_item_1->module_sack->p_impl->pool;
+    const Solvable * s1 = pool_id2solvable(pool, module_item_1->id.id);
+    const Solvable * s2 = pool_id2solvable(pool, module_item_2->id.id);
+    if (s1->name != s2->name) {
+        return s1->name < s2->name;
+    }
+    if (s1->arch != s2->arch) {
+        return s1->arch < s2->arch;
+    }
+    return module_item_1->get_version() > module_item_2->get_version();
+}
+
+
+void ModuleQuery::filter_latest(int limit) {
+    std::vector<const ModuleItem *> same_nsca_vector;
```

Comment:
Minor: I believe that the final size of `same_nsca_vector` (size of `same_nsca_vector.size()` == get_data().size() after filling in following for loop) is known, therefore you can reserve the space in the vector.